source/common/rate_limiter_impl.cc

```diff
@@ -118,4 +160,21 @@ bool FilteringRateLimiterImpl::tryAcquireOne() {
   return rate_limiter_->tryAcquireOne() ? filter_() : false;
 }
 
+GraduallyOpeningRateLimiterFilter::GraduallyOpeningRateLimiterFilter(
+    const std::chrono::nanoseconds ramp_time, DiscreteNumericDistributionSamplerPtr&& provider,
+    RateLimiterPtr&& rate_limiter)
+    : FilteringRateLimiterImpl(
+          std::move(rate_limiter),
+          [this]() {
+            if (elapsed() < ramp_time_) {
+              const double chance_percentage =
+                  100.0 - (static_cast<double>(ramp_time_.count() - elapsed().count()) /
+                           (ramp_time_.count() * 1.0)) *
+                              100.0;
+              return std::round(provider_->getValue() / 10000.0) <= chance_percentage;
```

Comment:
I am trying to understand the fixed division by 10k. It seems to imply some expectations about the discrete numeric distribution provider that don't seem to be specified on the API.

This may be related to my comment that we should add more documentation next to the constructor for the GraduallyOpeningRateLimiterFilter that will explain the arguments and their roles.